Detectors/ZDC/raw/src/RawReaderZDC.cxx

```diff
@@ -32,25 +32,50 @@ void RawReaderZDC::clear()
   mOrbitData.clear();
 }
 
-void RawReaderZDC::processBinaryData(gsl::span<const uint8_t> payload, int linkID)
+int RawReaderZDC::processBinaryData(gsl::span<const uint8_t> payload, int linkID, uint8_t dataFormat)
 {
   if (0 <= linkID && linkID < 16) {
     size_t payloadSize = payload.size();
-    for (int32_t ip = 0; ip < payloadSize; ip += 16) {
+    if (dataFormat == 2) {
+      for (int32_t ip = 0; (ip + PayloadPerGBTW) <= payloadSize; ip += PayloadPerGBTW) {
+        // Assign only the actual payload
+        uint32_t gbtw[4] = {0x0, 0x0, 0x0, 0x0};
+        memcpy((void*)gbtw, (const void*)&payload[ip], PayloadPerGBTW);
```

Comment:
Do I understand correctly that for the unpadded data you 1st copy every unpadded GBT word to padded version then decode it? It would be more natural to do the opposite: to extract the payload from the padded version and pass it to decoder expecting pure payload.

Reply:
Yes, it avoided to introduce more detailed pointer treatment. Now I removed the unnecessary copy.